mungegithub/mungers/triage-filer.go

```diff
@@ -0,0 +1,516 @@
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	githubapi "github.com/google/go-github/github"
+	"k8s.io/test-infra/mungegithub/features"
+	"k8s.io/test-infra/mungegithub/github"
+	"k8s.io/test-infra/mungegithub/mungers/mungerutil"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+)
+
+const (
+	timeFormat = "2 Jan 2006 15:04 MST"
+)
+
+// TriageFiler files issues for clustered flaky tests.
+type TriageFiler struct {
+	clusterDataURL                                string
+	topJobsCount, topTestsCount, topClustersCount int
+	triageURL                                     string
+	windowDays                                    int
+	syncFreq                                      time.Duration
+
+	nextSync    time.Time
+	latestStart int64
+
+	data    *triageData
+	config  *github.Config
+	creator *IssueCreator
+}
+
+func init() {
+	RegisterMungerOrDie(&TriageFiler{})
+}
+
+// Name returns the string identifier of this munger, usable in --pr-mungers flag.
+func (f *TriageFiler) Name() string { return "triagefiler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided for this munger to run.
+// The TriageFiler has no required features.
+func (f *TriageFiler) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the TriageFiler.
+// During initialization, the TriageFiler looks up the IssueCreator and sets the nextSync time.
+func (f *TriageFiler) Initialize(config *github.Config, features *features.Features) error {
+	f.config = config
+	f.nextSync = time.Now().Add(time.Duration(-1) * time.Minute) // Start off due for sync.
+
+	for _, munger := range GetActiveMungers() {
+		if munger.Name() == IssueCreatorName {
+			f.creator = munger.(*IssueCreator)
+		}
+	}
+	if f.creator == nil {
+		return fmt.Errorf("TriageFiler couldn't find the IssueCreator among active mungers.")
+	}
+
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop.
+// The TriageFiler does work only if it hasn't done so in the past syncFreq minutes.
+func (f *TriageFiler) EachLoop() error {
+	if time.Now().Before(f.nextSync) {
+		return nil
+	}
+
+	if err := f.FileIssues(); err != nil {
+		return err
+	}
+
+	f.nextSync = time.Now().Add(f.syncFreq)
+	glog.Infof("TriageFiler will run again at %v", f.nextSync)
+	return nil
+}
+
+// FileIssues is the main work function of the TriageFiler.  It fetches and parses cluster data,
+// then syncs the top issues to github with the IssueCreator.
+func (f *TriageFiler) FileIssues() error {
+	rawjson, err := mungerutil.ReadHTTP(f.clusterDataURL)
+	if err != nil {
+		return err
+	}
+	clusters, err := f.loadClusters(rawjson)
+	if err != nil {
+		return err
+	}
+	topclusters := topClusters(clusters, f.topClustersCount)
+	for _, clust := range topclusters {
+		f.creator.Sync(clust)
+	}
+	return nil
+}
+
+// AddFlags will add any requested flags to the cobra `cmd`.
+func (f *TriageFiler) AddFlags(cmd *cobra.Command, config *github.Config) {
```

Comment:
Thoughts:

* Is it important to start with these as flags rather than consts? It is cheap/trivial to release a new version.

* Should all of these start with triage? 

* Shouldn't the words be separated with hyphens? Aka `cluster-data-url` &c